### ISSUE TITLE
Issue when deploying example pod with VMSS User Assigned Managed Identity

### ISSUE THREAD

**Issue by lastcoolnameleft** [Azure/secrets-store-csi-driver-provider-azure#179]
**What steps did you take and what happened:**
https://github.com/Azure/secrets-store-csi-driver-provider-azure#deploy-your-kubernetes-resources

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: azure-kvname
spec:
  provider: azure
  parameters:
    usePodIdentity: "false"         # [OPTIONAL] if not provided, will default to "false"
    useVMManagedIdentity: "true"   # [OPTIONAL available for version > 0.0.4] if not provided, will default to "false"
    userAssignedIdentityID: "3f9cf9dd-7707-4a2a-a323-c255b0f1c2fa"
    keyvaultName: "msal-proxy"          # the name of the KeyVault
    objects:  |
      array:
        - |
          objectName: examplepasswd
          objectType: secret        # object types: secret, key or cert
          objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
    resourceGroup: "msal-proxy-rg"               # [REQUIRED for version < 0.0.4] the resource group of the KeyVault
    subscriptionId: "df8428d4-bc25-4601-b458-1c8533ceec0b"            # [REQUIRED for version < 0.0.4] the subscription ID of the KeyVault
    tenantId: "72f988bf-86f1-41af-91ab-2d7cd011db47"                    # the tenant ID of the KeyVau 
```

```
kind: Pod
apiVersion: v1
metadata:
  name: nginx-secrets-store-inline
spec:
  containers:
  - image: nginx
    name: nginx
    volumeMounts:
    - name: secrets-store-inline
      mountPath: "/mnt/secrets-store"
      readOnly: true
  volumes:
    - name: secrets-store-inline
      csi:
        driver: secrets-store.csi.k8s.io
        readOnly: true
        volumeAttributes:
          secretProviderClass: "azure-kvname"
        nodePublishSecretRef:
          name: secrets-store-creds
```

**What did you expect to happen:**
I expected the pod to successfully deploy, but instead got:
```
  Normal   Scheduled    3m7s                default-scheduler                           Successfully assigned default/nginx-secrets-store-inline to aks-nodepool1-32922832-vmss000002
  Warning  FailedMount  64s                 kubelet, aks-nodepool1-32922832-vmss000002  Unable to attach or mount volumes: unmounted volumes=[secrets-store-inline], unattached volumes=[secrets-store-inline default-token-fgqf2]: timed out waiting for the condition
  Warning  FailedMount  59s (x9 over 3m7s)  kubelet, aks-nodepool1-32922832-vmss000002  MountVolume.SetUp failed for volume "secrets-store-inline" : fetching NodePublishSecretRef default/secrets-store-creds failed: kubernetes.io/csi: failed to find the secret secrets-store-creds in the namespace default with error: secrets "secrets-store-creds" not found       [0.4s]
```

**Anything else you would like to add:**
The example nginx.yaml references secrets-store-creds but doesn't provide any detail about it, or even if it's an optional value.
It's unclear from the documentation, but I think that this is only valuable if your using SP identity because I found steps for it there `secrets-store-creds`; however, not in the User/System Assigned Managed Identity.

**Which access mode did you use to access the Azure Key Vault instance:**
User Assigned Managed Identity

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):us.gcr.io/k8s-artifacts-prod/csi-secrets-store/driver:v0.0.12
- Azure Key Vault provider version: (use the image tag): mcr.microsoft.com/k8s/csi/secrets-store/provider-azure:0.0.7
- Kubernetes version: (use `kubectl version` and `kubectl get nodes -o wide`):v1.16.10
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

**Comment by aramase** [comment]
@lastcoolnameleft Thank you for opening the issue 👍 

The `nodePublishSecretRef` is only required for ServicePrincipal mode - https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/docs/service-principal-mode.md. For other modes it's not required. It's documented in the service principal mode doc, but we'll also add info in the yaml.

**Comment by lastcoolnameleft** [comment]
Thanks for that hint.  I deleted it from my pod yaml and it worked (after deleting and re-creating the pod).  

I agree that something in the YAML to indicate which fields are necessary might help.

**Comment by asc-adean** [comment]
I am running into the same issue (using a managed identity, not a service principal).  I do not have the `nodePublishSecretRef` in my volume definition:

using helm:
`volumes:
        - name: secrets-store-inline
          csi:
            driver: secrets-store.csi.k8s.io
            readOnly: true
            volumeAttributes:
              secretProviderClass: {{ .Values.azure_keyvault.name }}`

**Comment by aramase** [comment]
@asc-adean What is the error you see in `kubectl describe` output for the pod? What is the kubernetes version (`kubectl get nodes -o wide` and `kubectl version`)? Could you provider sample of your `SecretProviderClass` yaml?

**Comment by asc-adean** [comment]
kubectl describe:
`Warning  FailedMount  3s  kubelet, aks-nodepool1-11007748-vmss000000  MountVolume.SetUp failed for volume "secrets-store-inline" : kubernetes.io/csi: mounter.SetupAt failed: rpc error: code = Unknown desc = error mounting secret time="2020-08-07T18:12:07Z" level=fatal msg="[error] : unexpected: getCredential failed, nodePublishSecretRef secret is not provided"`

nodes:
`aks-nodepool1-11007748-vmss000000   Ready    agent   2d    v1.16.10   172.21.0.4    <none>        Ubuntu 16.04.6 LTS   4.15.0-1089-azure   docker://3.0.10+azure`

kubectl version:
`Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-16T00:04:31Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.10", GitCommit:"89d8075525967c7a619641fabcb267358d28bf08", GitTreeState:"clean", BuildDate:"2020-06-23T02:52:37Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}`

sample yaml (sanitized):
`
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: client-azure-kvname
  namespace: sr-api
spec:
  provider: azure                   
  parameters:
    usePodIdentity: "false"         # [OPTIONAL for Azure] if not provided, will default to "false"
    useVMManagedIdentity: "false"   # [OPTIONAL available for version > 0.0.4] if not provided, will default to "false"
    userAssignedIdentityID: "xxxxx-2ec3-409f-8f37-xxxxxx"  # [OPTIONAL available for version > 0.0.4] use the client id to specify which user assigned managed identity to use. If using a user assigned identity as the VM's managed identity, then specify the identity's client id. If empty, then defaults to use the system assigned identity on the VM
    keyvaultName: "xxxx-dev"          # the name of the KeyVault
    objects:  |
      array:
        - |
          objectName: AppSettings--ConnectionString
          objectAlias: AppSettings__ConnectionString
          objectType: secret
          objectVersion: ""
    resourceGroup: "xxxx-dev-rg"
    subscriptionId: "xxxxx-a7de-4882-a937-xxxx"
    tenantId: "xxxx-3ee2-48f2-9d35-xxxx" 
`

**Comment by asc-adean** [comment]
Additionally, logs from the keyvault-secrets-store-csi-driver

```
time="2020-08-07T18:13:05Z" level=info msg="provider command invoked: /etc/kubernetes/secrets-store-csi-providers/azure/provider-azure --attributes [REDACTED] --secrets [REDACTED] [--targetPath /var/lib/kubelet/pods/6b895e63-2c8d-4127-9173-539cdfa03fe9/volumes/kubernetes.io~csi/secrets-store-inline/mount --permission 420]"
time="2020-08-07T18:13:05Z" level=info msg="time=\"2020-08-07T18:13:05Z\" level=info msg=\"not using pod identity or vm assigned user identity to access keyvault\"\ntime=\"2020-08-07T18:13:05Z\" level=info msg=\"missing client credential to access keyvault\"\n"
time="2020-08-07T18:13:05Z" level=error msg="error invoking provider, err: exit status 1, output: time=\"2020-08-07T18:13:05Z\" level=fatal msg=\"[error] : unexpected: getCredential failed, nodePublishSecretRef secret is not provided\"\n for pod: 6b895e63-2c8d-4127-9173-539cdfa03fe9, ns: sr-api"
time="2020-08-07T18:13:05Z" level=info msg="unmounting target path /var/lib/kubelet/pods/6b895e63-2c8d-4127-9173-539cdfa03fe9/volumes/kubernetes.io~csi/secrets-store-inline/mount as node publish volume failed"
time="2020-08-07T18:13:05Z" level=error msg="GRPC error: error mounting secret time=\"2020-08-07T18:13:05Z\" level=fatal msg=\"[error] : unexpected: getCredential failed, nodePublishSecretRef secret is not provided\"\n for pod: 6b895e63-2c8d-4127-9173-539cdfa03fe9, ns: sr-api"
```

**Comment by aramase** [comment]
@asc-adean If you're using managed identity, then `useVMManagedIdentity: "false"` needs to be set to true. As `useVMManagedIdentity: "false"` and `usePodIdentity: "false"`, the fallback option is to auth using `ServicePrincipal` which is why it's failing. 

Please set this to true for managed identity 

```yaml
    useVMManagedIdentity: "false"   # [OPTIONAL available for version > 0.0.4] if not provided, will default to "false"
```

**Comment by asc-adean** [comment]
Thanks @aramase I think I have it working now.
